lib/controllers/hosted-skill-controller/helper.js

```diff
@@ -67,15 +67,18 @@ function pollingSkillStatus(smapiClient, skillId, callback) {
         response,
       );
     }
+    
+    const empty = (data) => !data || R.isEmpty(data);
+    
     return (
-      !statusTracker.manifest ||
-      !statusTracker.interactionModel ||
-      !statusTracker.hostedSkillProvisioning ||
-      !(
-        statusTracker.manifest === CONSTANTS.HOSTED_SKILL.MANIFEST_STATUS.SUCCESS &&
-        statusTracker.interactionModel === CONSTANTS.HOSTED_SKILL.INTERACTION_MODEL_STATUS.SUCCESS &&
-        statusTracker.hostedSkillProvisioning === CONSTANTS.HOSTED_SKILL.PROVISIONING_STATUS.SUCCESS
-      )
+      // retry if one of manifest, interaction model, provisioning don't exist
+      empty(statusTracker.manifest) ||
+      empty(statusTracker.interactionModel) ||
+      empty(statusTracker.hostedSkillProvisioning) ||
+      // retry if one of manifest, interaction model, provisioning are in progress
+      statusTracker.manifest === CONSTANTS.HOSTED_SKILL.MANIFEST_STATUS.IN_PROGRESS ||
```

Comment:
I'm not sure if there was some maintainability reason behind separate constants/if they are subject to change, but I agree, it can be collapsed into a single constant. and this does look a lot cleaner! Will update in the next PR, missed your comment.

Reply:
oh no worries, it was minor, thanks for fixing this!